src/java/main/org/apache/zookeeper/server/quorum/Leader.java

```diff
@@ -584,6 +584,12 @@ void lead() throws IOException, InterruptedException {
                         }
                     }
 
+                    // check leader running status
+                    if (!this.isRunning()) {
+                        shutdown("Unexpected internal error");
```

Comment:
Yes, ZooKeeperServerListenerImpl will do error logging before setting server running status to ERROR and from that admin can determine the cause. I will change the log priority to "error" in ZooKeeperServerListenerImpl.java class. Is that OK?

LOG.error("Thread {} exits, error code {}", threadName, exitCode);